ggshield/core/git_shell.py

```diff
@@ -302,13 +308,14 @@ def tar_from_ref_and_filepaths(
 
     with tarfile.open(fileobj=tar_stream, mode="w:gz") as tar:
         for path in filepaths:
-            raw_file_content = git(["show", f"{ref}:./{path}"], cwd=wd)
 
-            if acceptation_func is not None and not (
-                acceptation_func(path, raw_file_content)
-            ):
+            def _readfile() -> str:
+                return read_git_file(ref, path, wd)
+
+            if acceptation_func is not None and not acceptation_func(path, _readfile):
                 continue
```

Comment:
Note: using `lru_cache` for this might be a waste.
Each filepath goes through the filter (1st read) then, if accepted, is immediately added to the tar (2nd read).
We go on to the next file and no longer need to open the previous one, so space is wasted in the cache).

However, this does not seem to impact performances.